# This patch file was generated by NetBeans IDE
# Following Index: paths are relative to: D:\solr\lucene_solr_3_1\lucene\src\test\org\apache\lucene\search\spans
# This patch can be applied using context Tools: Patch action on respective folder.
# It uses platform neutral UTF-8 encoding and \n newlines.
# Above lines and this line are ignored by the patching process.
Index: TestNearSpansOrdered.java
--- TestNearSpansOrdered.java Base (BASE)
+++ TestNearSpansOrdered.java Locally Modified (Based On LOCAL)
@@ -82,6 +82,21 @@
        slop,
        inOrder);
   }
+
+  protected SpanNearQuery makeOverlapedQuery() {
+      return new SpanNearQuery(
+              new SpanQuery[] {
+                new SpanNearQuery( new SpanQuery[] {
+                                    new SpanTermQuery(new Term(FIELD, "w3")),
+                                    new SpanTermQuery(new Term(FIELD, "w5")) },
+                                    1,
+                                    true
+                                ),
+                new SpanTermQuery(new Term(FIELD, "w4")) },
+                0,
+                true);
+  }
+
   protected SpanNearQuery makeQuery() {
     return makeQuery("w1","w2","w3",1,true);
   }
@@ -183,4 +198,9 @@
                0.0f < e.getValue());
   }
   
+  public void testOverlapedOrderedSpan()  throws Exception {
+    SpanNearQuery q = makeOverlapedQuery();
+    CheckHits.checkHits(random, q, FIELD, searcher, new int[] {});
 }
+
+}
